far/global.hpp

```diff
@@ -42,14 +42,58 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include "platform.hpp"
 
 // Common:
+#include "common/bytes_view.hpp"
 #include "common/preprocessor.hpp"
 
 // External:
 
 //----------------------------------------------------------------------------
 
-enum class lng;
-enum class search_case_fold;
+struct global_search_replace_params
+{
+	NONCOPYABLE(global_search_replace_params);
+	NONMOVABLE(global_search_replace_params);
+
+	global_search_replace_params() = default;
+
+						// It is client's responsibility to maintain invariants:
+	string SearchStr;   // If Hex, should be BytesToString(SearchBytes, CodePage)
+	bytes SearchBytes;  // If not Hex, should be empty
+	string ReplaceStr;
+	bool Hex{};
+	bool CaseSensitive{};
+	bool WholeWords{};
+	bool Regex{};
+	bool Fuzzy{};
+	bool PreserveStyle{};
+};
+
+template <typename client_params_t>
+void operator<<(global_search_replace_params& GlobalParams, const client_params_t& ClientParams)
```

Comment:
Mostly because the semantics does not fit well with the conventional meaning of assignment. Another, less important consideration was to discourage accidental unintentional assignment. Maybe making it a named function is a better idea, especially if we want to pass in the code page. What do you think?

Reply:
I'd probably prefer a named function (save, store, persist etc.) if it does not fit well.
Overloading operators is great for vocabulary types, global_search_replace_params is unlikely one of them.

Reply:
Done.